From 1d18591d9e7ed9b3580e06c37be23bfde24bc30c Mon Sep 17 00:00:00 2001
From: Peter Seiderer <ps.report@gmx.net>
Date: Fri, 3 Apr 2020 22:40:29 +0200
Subject: [PATCH] examples: fix wayland/texture-sharing/custom-compositor
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fixes:

  main.cpp:83:33: error: ‘GL_RGBA8’ was not declared in this scope; did you mean ‘GL_RGBA4’?
     83 |             *glInternalFormat = GL_RGBA8;
        |                                 ^~~~~~~~
        |                                 GL_RGBA4
  main.cpp:120:33: error: ‘GL_RGBA8’ was not declared in this scope; did you mean ‘GL_RGBA4’?
    120 |             *glInternalFormat = GL_RGBA8;
        |                                 ^~~~~~~~
        |                                 GL_RGBA4

Task-number: QTBUG-83304
Change-Id: I0515f3303b08c405d162986e280e9a7bd35e28f7
Reviewed-by: Laszlo Agocs <laszlo.agocs@qt.io>
Reviewed-by: Eskil Abrahamsen Blomfeldt <eskil.abrahamsen-blomfeldt@qt.io>

Upstream-Status: Backport [1d18591d9e7ed9b3580e06c37be23bfde24bc30c]

The patch is backport from upstream commit -
1d18591d9e7ed9b3580e06c37be23bfde24bc30c

Signed-off-by: Gowtham Tammana <g-tammana@ti.com>
---
 examples/wayland/texture-sharing/custom-compositor/main.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/examples/wayland/texture-sharing/custom-compositor/main.cpp b/examples/wayland/texture-sharing/custom-compositor/main.cpp
index a39c8c38..af375024 100644
--- a/examples/wayland/texture-sharing/custom-compositor/main.cpp
+++ b/examples/wayland/texture-sharing/custom-compositor/main.cpp
@@ -63,6 +63,10 @@
 
 #include "QtWaylandCompositor/private/qwltexturesharingextension_p.h"
 
+#ifndef GL_RGBA8
+#define GL_RGBA8 0x8058
+#endif
+
 class CustomSharingExtension : public QWaylandTextureSharingExtension
 {
     Q_OBJECT
-- 
2.17.1

